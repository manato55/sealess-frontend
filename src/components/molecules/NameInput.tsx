import React, { Dispatch, SetStateAction } from 'react';
import Input from '../atoms/Input';
import ErrorMessageWrapper from '../atoms/ErrorMessageWrapper';
import { useRecoilValue } from 'recoil';
import { authErrorMessage, eachErrorFlag } from '../../store/atom';

interface Props {
  setName: Dispatch<SetStateAction<string>>;
  name: string;
}

export const FullName = (props: Props) => {
  const errorMessage = useRecoilValue(authErrorMessage);
  const errorFlag = useRecoilValue(eachErrorFlag);

  return (
    <>
      <ErrorMessageWrapper>{errorFlag.name && errorMessage.name[0]}</ErrorMessageWrapper>
      <Input type="text" value={props.name} onChange={(e: React.ChangeEvent<HTMLInputElement>) => props.setName(e.target.value)} placeholder="氏名" />
    </>
  );
};

export default FullName;
