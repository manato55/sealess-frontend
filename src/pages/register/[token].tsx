import React, { useEffect, useState, useCallback } from 'react';
import { useRouter } from 'next/router';
import { useAuthenticate } from '../../hooks/useAuthenticate';
import Password from '../../components/molecules/Password';
import Button from '../../components/atoms/Button';
import AuthWrapper from '../../components/atoms/AuthWrapper';

export const Token = (): React.ReactElement => {
  const router = useRouter();
  const { tokenCheck } = useAuthenticate();
  const { officialRegistryForOrdinaryUser } = useAuthenticate();
  const [paramsToken, setParamsToken] = useState<any>(router.query.token);
  const [tokenChecker, setTokenChecker] = useState<boolean>(false);
  const [password, setPassword] = useState<string>('');

  useEffect(() => {
    const initialAction = async () => {
      const res = await tokenCheck(paramsToken);
      if (res) {
        setTokenChecker(true);
      }
    };
    initialAction();
  }, [paramsToken]);

  async function submitRegister(): Promise<void> {
    const data: {
      token: string;
      password: string;
    } = {
      token: paramsToken,
      password: password,
    };
    await officialRegistryForOrdinaryUser(data);
  }

  return (
    <>
      {tokenChecker ? (
        <AuthWrapper>
          <Password password={password} setPassword={setPassword} />
          <Button onClick={() => submitRegister()}>登録</Button>
        </AuthWrapper>
      ) : (
        'Loading...'
      )}
    </>
  );
};

export default Token;
