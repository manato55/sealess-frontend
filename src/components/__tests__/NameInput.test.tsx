import React from 'react';
import NameInput from '../molecules/NameInput';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';

describe('NameInput test', () => {
  let container;
  const onChangeMock = jest.fn();

  beforeEach(() => {
    container = mount(
      <RecoilRoot>
        <NameInput setName={onChangeMock} name={'foo'} />
      </RecoilRoot>,
    );
  });

  test('input test', () => {
    const input = container.find('input');
    input.simulate('change');
    expect(onChangeMock).toHaveBeenCalledWith('foo');
  });

  test('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
