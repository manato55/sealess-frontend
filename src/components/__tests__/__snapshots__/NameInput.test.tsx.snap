// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameInput test snapshot 1`] = `
<RecoilRoot>
  <RecoilRoot_INTERNAL>
    <Batcher
      setNotifyBatcherOfChange={[Function]}
    />
    <FullName
      name="foo"
      setName={
        [MockFunction] {
          "calls": Array [
            Array [
              "foo",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <styled.p>
        <p
          className="sc-gtsrHT DoTxJ"
        />
      </styled.p>
      <styled.input
        editable={true}
        onChange={[Function]}
        placeholder="氏名"
        type="text"
        value="foo"
        visible={false}
      >
        <input
          className="sc-bdnxRM elxLYl"
          onChange={[Function]}
          placeholder="氏名"
          type="text"
          value="foo"
        />
      </styled.input>
    </FullName>
  </RecoilRoot_INTERNAL>
</RecoilRoot>
`;
